src/core/qgsrenderedlayerstatistics.h

```diff
@@ -0,0 +1,133 @@
+    copyright            : (C) 2024 by Jean Felder
+    email                : jean dot felder at oslandia dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSRENDEREDLAYERSTATISTICS_H
+#define QGSRENDEREDLAYERSTATISTICS_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsrendereditemdetails.h"
+
+#include <limits>
+
+
+/**
+ * \ingroup core
+ * \brief Contains computed statistics for a layer render.
+ * \since QGIS 3.42
+ */
+class CORE_EXPORT QgsRenderedLayerStatistics: public QgsRenderedItemDetails
+{
+  public:
+
+    /**
+     * Constructor for QgsRenderedLayerStatistics from a list of \a minimum and \a maximum.
+     */
+    QgsRenderedLayerStatistics( const QString &layerId, QList<double> minimum, QList<double> maximum );
+
+    /**
+     * Constructor for QgsRenderedLayerStatistics with only one value for \a minimum and \a maximum.
+     */
+#ifndef SIP_RUN
+    QgsRenderedLayerStatistics( const QString &layerId, double minimum = std::numeric_limits<double>::quiet_NaN(), double maximum = std::numeric_limits<double>::quiet_NaN() );
+#else
+    QgsRenderedLayerStatistics( const QString &layerId, SIP_PYOBJECT minimum SIP_TYPEHINT( Optional[float] ) = Py_None, SIP_PYOBJECT maximum SIP_TYPEHINT( Optional[float] ) = Py_None ) [( const QString & layerId, double minimum = 0.0, double maximum = 0.0 )];
+    % MethodCode
+    double minP = a1 == Py_None ? std::numeric_limits<double>::quiet_NaN() : PyFloat_AsDouble( a1 );
+    double maxP = a2 == Py_None ? std::numeric_limits<double>::quiet_NaN() : PyFloat_AsDouble( a2 );
+    QList<double> minL = {minP};
+    QList<double> maxL = {maxP};
+    sipCpp = new sipQgsRenderedLayerStatistics( *a0, minL, maxL );
+    % End
+#endif
+
+#ifdef SIP_RUN
+    SIP_PYOBJECT __repr__();
+    % MethodCode
+    QStringList minimums;
+    minimums.reserve( sipCpp->minimum().size() );
+    for ( double min : sipCpp->minimum() )
+    {
+      minimums.append( QString::number( min ) );
+    }
+    QStringList maximums;
+    maximums.reserve( sipCpp->maximum().size() );
+    for ( double max : sipCpp->maximum() )
+    {
+      maximums.append( QString::number( max ) );
+    }
+    QString str = QStringLiteral( "<QgsRenderedLayerStatistics: %1 ([%2] x [%3])>" ).arg( sipCpp->layerId(), minimums.join( ',' ), maximums.join( ',' ) );
```

Comment:
"[ ] x [ ]" is an odd choice here, that looks a bit like a matrix of values. Maybe just explicitly go "min: %2, max %3" instead?

Reply:
Done.